File: backend/app/schemas/account_config.py
"""Account configuration schemas."""
from typing import Optional, List, Any
from datetime import datetime
from pydantic import BaseModel


class AccountConfigBase(BaseModel):
    """Account configuration base schema."""
    account_id: str
    account_name: Optional[str] = None
    initial_capital: float = 0.0
    currency: str = "CNY"
    broker: Optional[str] = None
    description: Optional[str] = None
    tags: List[str] = []
    config: dict = {}


class AccountConfigCreate(AccountConfigBase):
    """Account configuration create schema."""
    pass


class AccountConfigUpdate(BaseModel):
    """Account configuration update schema."""
    account_name: Optional[str] = None
    initial_capital: Optional[float] = None
    currency: Optional[str] = None
    broker: Optional[str] = None
    description: Optional[str] = None
    tags: Optional[List[str]] = None
    config: Optional[dict] = None


class AccountConfigResponse(AccountConfigBase):
    """Account configuration response schema."""
    created_at: datetime
    updated_at: Optional[datetime] = None
    created_by: Optional[int] = None

    class Config:
        from_attributes = True


class AccountPermissionBase(BaseModel):
    """Account permission base schema."""
    user_id: int
    account_id: str
    permission_type: str = "view"


class AccountPermissionUpdate(AccountPermissionBase):
    """Account permission update schema."""
    pass


class AccountPermissionResponse(AccountPermissionBase):
    """Account permission response schema."""
    id: int
    created_at: datetime
    created_by: Optional[int] = None

    class Config:
        from_attributes = True


class AccountPermissionDetail(BaseModel):
    """Enhanced account permission with user and account details."""
    id: int
    user_id: int
    username: str
    account_id: str
    account_name: Optional[str] = None
    permission_type: str
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/services/accountConfig.ts
import api from './api';

export interface AccountConfig {
  account_id: string;
  account_name?: string;
  initial_capital: number;
  currency: string;
  broker?: string;
  description?: string;
  tags: string[];
  created_at: string;
  updated_at?: string;
  created_by?: number;
}

export interface AccountConfigCreate {
  account_id: string;
  account_name?: string;
  initial_capital: number;
  currency?: string;
  broker?: string;
  description?: string;
  tags?: string[];
  config?: Record<string, any>;
}

export interface AccountConfigUpdate {
  account_name?: string;
  initial_capital?: number;
  currency?: string;
  broker?: string;
  description?: string;
  tags?: string[];
  config?: Record<string, any>;
}

export interface AccountPermission {
  id: number;
  user_id: number;
  account_id: string;
  permission_type: string;
  created_at: string;
  created_by?: number;
}

export interface AccountPermissionUpdate {
  user_id: number;
  account_id: string;
  permission_type: string;
}

export const accountConfigApi = {
  // 获取所有交易账户（管理员）
  getAllAccounts: () =>
    api.get<AccountConfig[]>('/account-config/accounts'),

  // 创建交易账户（管理员）
  createAccount: (data: AccountConfigCreate) =>
    api.post<AccountConfig>('/account-config/accounts', data),

  // 更新交易账户（管理员）
  updateAccount: (accountId: string, data: AccountConfigUpdate) =>
    api.put<AccountConfig>(`/account-config/accounts/${accountId}`, data),

  // 删除交易账户（管理员）
  deleteAccount: (accountId: string) =>
    api.delete(`/account-config/accounts/${accountId}`),

  // 获取权限矩阵（管理员）
  getPermissions: () =>
    api.get<AccountPermission[]>('/account-config/permissions'),

  // 更新权限分配（管理员）
  updatePermissions: (permissions: AccountPermissionUpdate[]) =>
    api.put('/account-config/permissions', permissions),

  // 获取我的交易账户（所有用户）
  getMyAccounts: () =>
    api.get<AccountConfig[]>('/account-config/my-accounts'),

  // 获取特定用户的权限
  getUserPermissions: (userId: number) =>
    api.get<string[]>(`/account-config/permissions/user/${userId}`),

  // 设置用户权限
  setUserPermissions: (userId: number, accountIds: string[]) =>
    api.post(`/account-config/permissions/user/${userId}`, accountIds),
};